### PR TITLE
drivers: pwm: fix esp32-s3 ledc pwm low frequency range

### PR DESCRIPTION
**Pull request by jkandasa**
### Issues:
LEDC PWM not working with very low frequency (lesser than 30Hz).

### Root cause:
* The plain ESP32 variant has 20 bits counter and can work with as lower as possible (wiht 1Hz).
* However the ESP32 S2,S3 and C3 variants has only 14 bits counter, so it failed to go below 30Hz.

### Fix:
* If `pwm_led_esp32_calculate_max_resolution` can not resolve the given lower frequency, just set the least possible frequency option.
* This will be further tuned with `clock divider` on `pwm_led_esp32_timer_set` function with `prescaler`.

### CODE REVIEW

**Comment by jkandasa** [comment]
@LucasTambor @sylvioalves Can you please have look on this?

**Comment by kartben** [comment]
@jkandasa in the meantime you may want to fix the commit message

```
 5: UC4 Commit message body line exceeds max length (76>75): "application failed to set very low frequency(1Hz) in S2, S3 and C3 variants."
9: UC4 Commit message body line exceeds max length (80>75): "this will be handled on the function 'pwm_led_esp32_timer_set' with 'prescaler'."
```

**Comment by jkandasa** [comment]
@kartben I fixed the lint issue(max line on comments), but still it is failing with previous commit. 
I have rebased and force pushed my changes

**Comment by kartben** [comment]
> @kartben I fixed the lint issue(max line on comments), but still it is failing with previous commit. 
> I have rebased and force pushed my changes

You need to amend the commit message so that it wraps at 75 characters, and then force push

File: drivers/pwm/pwm_led_esp32.c [hunk 1]
```diff
@@ -169,7 +169,18 @@ static int pwm_led_esp32_timer_config(struct pwm_ledc_esp32_channel_config *chan
...
+	/**
+	 * ESP32 - S2,S3 and C3 variants have only 14 bits counter.
+	 * where as the plain ESP32 variant has 20 bits counter.
+	 * application failed to set low frequency(1Hz) in S2, S3 and C3 variants.
+	 * to get very low frequencies on these variants,
+	 * frequency needs to be tuned with 18 bits clock divider.
+	 * so select the slow clock source (1MHz) with highest counter resolution.
+	 * this can be handled on the func 'pwm_led_esp32_timer_set' with 'prescaler'.
+	 */
+	channel->clock_src = LEDC_REF_TICK;
```
Comment (LucasTambor) [inline drivers/pwm/pwm_led_esp32.c:181]: This line is redundant.

**Review by LucasTambor — CHANGES_REQUESTED** [review]
Thanks @jkandasa, just a minor change request. Otherwise great job.

File: drivers/pwm/pwm_led_esp32.c (hunk 1 above)
Comment (jkandasa) [reply drivers/pwm/pwm_led_esp32.c]: @LucasTambor thanks for the review comment.
I removed that duplicate line.

**Review by sylvioalves — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

**Comment by jkandasa** [comment]
@kartben May I know the next step on this PR?
I see it is approved, but still not get merged. Has someone to merge the PR manually?

**Comment by kartben** [comment]
> @kartben May I know the next step on this PR? I see it is approved, but still not get merged. Has someone to merge the PR manually?

@jkandasa sure! It just needs to be re-approved by the assignee, it shouldn't take long :)

**Comment by jkandasa** [comment]
@kartben I do not know, why the GH action `Compliance Checks / Run compliance checks on patch series (PR) (pull_request)` failing again :cry:

**Comment by henrikbrixandersen** [comment]
> @kartben I do not know, why the GH action `Compliance Checks / Run compliance checks on patch series (PR) (pull_request)` failing again 😢

Simply click the "Details" link next to the check and you will see the following explanation:

Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines for more details

 5: UC4 Commit message body line exceeds max length (76>75): "application failed to set very low frequency(1Hz) in S2, S3 and C3 variants."
9: UC4 Commit message body line exceeds max length (80>75): "this will be handled on the function 'pwm_led_esp32_timer_set' with 'prescaler'."

**Comment by jkandasa** [comment]
@kartben thanks for your patience.
I was thinking about the lines inside files, totally forgot about the commit message and commit message description

**Comment by jkandasa** [comment]
@sylvioalves can you please re-approve the PR? Thanks!

**Comment by kartben** [comment]
As you haven't changed actual code as part of fixing the compliance issue, existing reviews are still valid. Expect the PR to be merged shortly by one of our release engineers :)

**Pull request merged**